schema/spacewalk/upgrade/susemanager-schema-5.1.0-to-susemanager-schema-5.1.1/003-virt-removal.sql

```diff
@@ -0,0 +1,36 @@
+--
+-- Copyright (c) 2024 SUSE LLC
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+DROP TABLE IF EXISTS rhnVirtualInstanceEventLog,
+                     rhnVirtualInstanceEventType,
+                     rhnActionVirtCreate,
+                     rhnActionVirtDelete,
+                     rhnActionVirtDestroy,
+                     rhnActionVirtMigrate,
+                     rhnActionVirtNetworkCreate,
+                     rhnActionVirtNetworkStateChange,
+                     rhnActionVirtPoolCreate,
+                     rhnActionVirtPoolDelete,
+                     rhnActionVirtPoolRefresh,
+                     rhnActionVirtPoolStart,
+                     rhnActionVirtPoolStop,
+                     rhnActionVirtReboot,
+                     rhnActionVirtRefresh,
+                     rhnActionVirtResume,
+                     rhnActionVirtSchedulePoller,
+                     rhnActionVirtSetMemory,
+                     rhnActionVirtShutdown,
+                     rhnActionVirtStart,
+                     rhnActionVirtSuspend,
+                     rhnActionVirtVcpu,
+                     rhnActionVirtVolDelete CASCADE;
+
+DELETE FROM rhnActionType WHERE label LIKE 'virt.%';
```

Comment:
Changing the history for actions that are supposed to not be used shouldn't be a problem. If that is a problem, it probably mean we need to keep the whole feature because someone uses it.

Reply:
It's a tricky one from legal perspective. Ideally, we should not remove any history from the systems, at least for a pre-defined time(6 months or so, different companies have different policies)

If we can somehow keep the history while removing the feature itself, that would be ideal. If that's not feasible and requires a lot more work, I would just go ahead with this change. However, would add a note in release notes that this will happen.

Reply:
The problem is that we will have to keep a dozen of tables and the matching Java code, just to show empty content because no-one was using it.

Reply:
Ok if there is no easy way, we go ahead with your change. We will add a note in release notes when times comes.